### PR TITLE
LINK-91, LINK-92 | Event form image section

### PR DESCRIPTION
**Pull request by jorilindell**
## Description 
Image selector component to select existing image
Image uploader component to upload images to LE. File can be added by drag'n'drop and using select file dialog  
Modal to add image to event
Fields to edit image details. Saving changes is not implemented in scope of this PR
Image section of the event form

## Closes
[LINK-91](https://helsinkisolutionoffice.atlassian.net/browse/LINK-91)
[LINK-92](https://helsinkisolutionoffice.atlassian.net/browse/LINK-92)

## Screenshots
Screenshot 2020-12-09 at 12 16 49

Screenshot 2020-12-09 at 12 17 35

Screenshot 2020-12-09 at 12 18 32

## Testing 
Selecting existing can be tested locally or on https://linkedcomponents-ui-qa-r-feature-imag-vthzdq.test.kuva.hel.ninja/fi/event/create?tab=8
Uploading images by file or url can only tested locally using API key authentication method. Note that you have to add 
```
CORS_ALLOW_HEADERS = [
    'apikey',
    'authorization',
    'content-type',
    'dnt',
    'origin',
    'user-agent',
    'x-csrftoken',
    'x-requested-with',
]
```   
to Linked Events API settings to get rid of CORS issues.

### CODE REVIEW

File: src/common/components/imageUploader/ImageUploader.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    /* istanbul ignore else  */
+    if (e.key === 'Enter') {
+      fileInput.current?.click();
+    }
+  };
+
+  return (
+    <div className={styles.imageUploader}>
+      <label
```
Comment (Sartonon) [inline src/common/components/imageUploader/ImageUploader.tsx:95]: My screen reader doesn't read this label 🤔

File: src/domain/app/apollo/apolloClient.ts [hunk 2]
```diff
@@ -1,31 +1,71 @@
...
   addTypenameEvent,
+  addTypenameImage,
   addTypenameKeyword,
   addTypenameKeywordSet,
   addTypenameLanguage,
   addTypenameMeta,
   addTypenamePlace,
 } from './utils';
 
+const uploadImageSerializer = (
```
Comment (Sartonon) [inline src/domain/app/apollo/apolloClient.ts:34]: Maybe this could use small description why this is needed? what do you think?

Comment (jorilindell) [reply src/domain/app/apollo/apolloClient.ts:38]: Added a short description

File: src/common/components/imageUploader/ImageUploader.tsx (hunk 1 above)
Comment (jorilindell) [reply src/common/components/imageUploader/ImageUploader.tsx]: svg image as a first label element caused the problem. Changed label to button to also get rid of input id generation

**Comment by Sartonon** [comment]
Looking good. Just couple comments to take a look at :)

File: src/domain/event/formSections/imageSection/ImageSection.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    name: EVENT_FIELDS.IMAGES,
+  });
+
+  const { refetch: refetchImages } = useImagesQuery({
+    skip: true,
+    variables: {
+      createPath: getPathBuilder(imagesPathBuilder),
+      pageSize: IMAGES_PAGE_SIZE,
+    },
+  });
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/formSections/imageSection/ImageSection.tsx]: Changed to useImagesLazyQuery, but to be honest I'm not sure which one is better or is there even much difference.

**Comment by Sartonon** [comment]
Great!

**Review by Sartonon — APPROVED** [review]

**Pull request merged**